### PR TITLE
Header escape

### PR DESCRIPTION
**Pull request by kum-deepak**

### CODE REVIEW

File: tests/unit/frame.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
 ﻿QUnit.module("Stomp Frame");
 
+QUnit.test("escape header value", function (assert) {
+  var out = Stomp.Frame.frEscape("anything\\a\nb\nc\rd\re:f:\\anything\\a\nb\nc\rd\re:f:\\");
+  assert.equal(out, "anything\\\\a\\nb\\nc\\rd\\re\\cf\\c\\\\anything\\\\a\\nb\\nc\\rd\\re\\cf\\c\\\\");
+});
+
+QUnit.test("escape header value", function (assert) {
+  var orig = "anything\\a\nb\nc\rd\re:f:\\anything\\a\nb\nc\rd\re:f:\\";
+  var out = Stomp.Frame.frUnEscape(Stomp.Frame.frEscape(orig));
```
Comment (Toparvion) [inline tests/unit/frame.js]: Not sure it is a good idea to rely on a function `frEscape` which is a SUT in sibling test. It introduces undesirable dependencies between tests and thus may cause failure propagation among them. I think it's better to verify with statically specified value just like in the test above.

File: tests/unit/frame.js [hunk 2]
```diff
@@ -1,5 +1,16 @@
 ﻿QUnit.module("Stomp Frame");
 
+QUnit.test("escape header value", function (assert) {
+  var out = Stomp.Frame.frEscape("anything\\a\nb\nc\rd\re:f:\\anything\\a\nb\nc\rd\re:f:\\");
+  assert.equal(out, "anything\\\\a\\nb\\nc\\rd\\re\\cf\\c\\\\anything\\\\a\\nb\\nc\\rd\\re\\cf\\c\\\\");
+});
+
+QUnit.test("escape header value", function (assert) {
```
Comment (Toparvion) [inline tests/unit/frame.js]: Typo: 
> [un]escape header value

File: src/stomp.coffee [hunk 3]
```diff
@@ -126,26 +133,38 @@ class Frame
...
+    frame = new Frame(command, headers, body, escapeHeaderValues)
     return frame.toString() + Byte.NULL
 
+  # Escape header values
+  #
+  # @private
+  @frEscape: (str) ->
+    ("" + str).replace(/\\/g, "\\\\").replace(/\r/g, "\\r").replace(/\n/g, "\\n").replace(/:/g, "\\c")
+
+  # Escape header values
```
Comment (Toparvion) [inline src/stomp.coffee]: Typo:

> [un]Escape header values

File: tests/unit/frame.js (hunk 1 above)
Comment (kum-deepak) [reply tests/unit/frame.js]: Actually it is intentional. The first test case tests that escape works (against a fixed expected output). The second one - having verified that escape works - tests that escape followed by unescape gives back the original and in the process verifies unescape.

There are other tests added as well which tests these at higher level.

**Comment by kum-deepak** [comment]
I will wait for few days then merge and release a new version.

File: tests/unit/frame.js (hunk 1 above)
Comment (Toparvion) [reply tests/unit/frame.js]: Ok, thanks for clarifying.

**Comment by Toparvion** [comment]
Slightly off-topic but still related question: do you plan to release your library branch as webjar like original library was (e.g. on Maven Central)? It would greatly help its adoption in JVM world.

**Comment by kum-deepak** [comment]
I actually don't know how it is done. Will you be able to help?

**Comment by Toparvion** [comment]
Yes, I think. I already have a project there. It is a plain library jar (not webjar) but I think there shouldn't be significant difference.

**Comment by Toparvion** [comment]
I've observed the [process of webjar publishing](https://www.webjars.org/contributing) and realized that it is completely different from the one of usual Java libraries (none the less). It doesn't seem complicated at all but involves remarkable amount of owner-specific data so that it would make sense to do it by yourself in order to avoid excess delegating.

**Comment by kum-deepak** [comment]
Actually I published the Webjars, or at least I think so. Please search for stompjs at http://www.webjars.org/

I used the option `Add a WebJar` on the left of the search box and used the npm option.

I will find a way to automate it so that I can hook it to the npm release process.

**Comment by Toparvion** [comment]
Yes, you're right - there is `stomp__stompjs` artifact appeared in Maven Central repository. You can check it at:
- http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22stomp__stompjs%22 or
- http://mvnrepository.com/artifact/org.webjars.npm/stomp__stompjs/3.1.1.

I've tried to include the library as a webjar into my Gradle project with these two lines of code:
1. `compile group: 'org.webjars.npm', name: 'stomp__stompjs', version: '3.1.1'` in Gradle script
2. and `<script src="webjars/stomp__stompjs/3.1.1/lib/stomp.js" type="text/javascript"></script>` in web page (intentionally ignored minified version)

and it works. Thank you!

I don't actually understand the difference between the simple releasing way you've found and the one I 've mentioned above. Nevertheless the result is good enough not to waste time on it anymore. 

I think you can include the first line (or its Maven analog) into the documentation as an alternative for downloading flat files. It is the most commonly adopted way of dependency resolution in (especially large) JVM applications.

**Pull request merged**